### ISSUE TITLE
Exception [Microsoft.AspNetCore.Components.Authorization.AuthorizeRouteView] for Published project

### ISSUE THREAD

**Issue by mwagil** [dotnet/aspnetcore#21475]
### Describe the bug

Running the project with Visual Studio and IIS Express works perfectly, but when I publish the project to a Full IIS or execute it using  the command "dotnet run" it throws the Exception bellow.

I already tried all the sugestions, with no success, found here: https://github.com/dotnet/aspnetcore/issues/18733

![image](https://user-images.githubusercontent.com/60952971/81009223-9bb01400-8e2a-11ea-91de-200c09b4ea7a.png)

**Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Default constructor not found for type Microsoft.AspNetCore.Components.Authorization.AuthorizeRouteView
System.MissingMethodException: Default constructor not found for type Microsoft.AspNetCore.Components.Authorization.AuthorizeRouteView
  at System.RuntimeType.CreateInstanceMono (System.Boolean nonPublic, System.Boolean wrapExceptions) <0x1c3dbe0 + 0x000b0> in <filename unknown>:0 
  at System.RuntimeType.CreateInstanceSlow (System.Boolean publicOnly, System.Boolean wrapExceptions, System.Boolean skipCheckThis, System.Boolean fillCache)**

### To Reproduce
Create a new project like:
![image](https://user-images.githubusercontent.com/60952971/81009421-f9446080-8e2a-11ea-81f3-98f136c2f531.png)

Add to Blazor Client Program.cs the lines:
```
builder.Services.AddScoped<AuthenticationStateProvider, AuthenticationService>();
builder.Services.AddAuthorizationCore();
```

Modify the App.razor file like this (included the AuthorizeRouteView):
    
```
<Router AppAssembly="@typeof(Program).Assembly">
        <Found Context="routeData">
            <AuthorizeRouteView RouteData="@routeData" DefaultLayout="@typeof(MainLayout)"/>
        </Found>
        <NotFound>
            <LayoutView Layout="@typeof(MainLayout)">
                <p>Sorry, there's nothing at this address.</p>
            </LayoutView>
        </NotFound>
    </Router>
```

Create the custom class AuthenticationService like:
```
public class AuthenticationService : AuthenticationStateProvider
    {
        public override async Task<AuthenticationState> GetAuthenticationStateAsync()
        {
            var claimsPrincipal = new ClaimsPrincipal(new ClaimsIdentity());

            return await Task.FromResult(new AuthenticationState(claimsPrincipal));
        }

    }
```

### Further technical details
- Visual Studio 16.6.0 Preview 5.0
- Blazor 3.2.0 rc1.20223.4
- .Net core 3.1.3
- Windows 10

**Comment by javiercn** [comment]
@mwagil thanks for contacting us.

You have a linker issue I believe. The problem is that the linker is stripping away the constructor. You can provide your own descriptor file to fix it. See [here](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/blazor/configure-linker?view=aspnetcore-3.1) for docs on how to do it.

**Comment by mwagil** [comment]
@javiercn thank you for the reply.

I tried the link you sent me, but it didn't fix, I kept trying to find what was causing this issue in my project and as desperate solution I did this:

- Closed the solution in Visual Studio
- Deleted the folder .vs
- Deleted the folders: Bin and Obj of each project inside the Solution.
- Opened the solution again in Visual Studio
- Deployed the project and voiala, it worked! :)

The sad part is, I don't know exactly what I did in first place to break my project, anyway, now it is working again.

Thank you for the attention.
